library/src/main/java/com/getstream/sdk/chat/ChatImpl.java

```diff
@@ -109,8 +118,46 @@ public String getVersion() {
         return BuildConfig.BUILD_TYPE + ":" + BuildConfig.VERSION_NAME;
     }
 
-    void init() {
+    @Override
+    public void setUser(@NotNull User user,
+                        @NotNull String userToken,
+                        @NotNull InitConnectionListener callbacks) {
+        disconnectChatDomainIfAlreadyInitialized();
+        final ChatClient client = new ChatClient.Builder(this.apiKey, context).build();
+        final ChatDomain.Builder domainBuilder = new ChatDomain.Builder(context, client, user);
+        if (offlineEnabled) {
+            domainBuilder.offlineEnabled();
+        }
+        final ChatDomain chatDomain = domainBuilder.userPresenceEnabled().build();
+        client.setUser(user, userToken, new InitConnectionListener() {
+            @Override
+            public void onSuccess(@NotNull ConnectionData data) {
+                chatDomain.setCurrentUser(user);
+                callbacks.onSuccess(data);
+            }
+
+            @Override
+            public void onError(@NotNull ChatError error) {
+                callbacks.onError(error);
+            }
+        });
+
+        init();
+    }
+
+    private void disconnectChatDomainIfAlreadyInitialized() {
+        try {
+            final ChatDomain chatDomain = ChatDomain.instance();
+            BuildersKt.launch(GlobalScope.INSTANCE,
+                    Dispatchers.getIO(),
+                    CoroutineStart.DEFAULT,
+                    (scope, continuation) -> chatDomain.disconnect(continuation));
+        } catch(UninitializedPropertyAccessException e) {
+            Log.d("ChatImpl", "ChatDomain was not initialized yet. No need to disconnect.");
```

Comment:
Client has logger. Better to use it for development logs, so customers could disable it, intercept it or filter by common tag prefix: `adb logcat | grep "Chat:"`
```kotlin
ChatLogger.get("tag").logD("log message")
```